tests/pebble.rs

```diff
@@ -296,6 +283,62 @@ impl PebbleEnvironment {
     }
 }
 
+/// A trait for something able to provision a challenge response with an external system
+#[async_trait]
+trait ChallengeProvisioner {
+    /// Provision a challenge response for the given identifier, challenge, and key auth.
+    async fn provision(
+        &self,
+        identifier: String,
+        challenge: &Challenge,
+        key_auth: &KeyAuthorization,
+    ) -> Result<(), Box<dyn StdError>>;
+}
+
+/// Provisions an HTTP-01 challenge response.
+///
+/// The Pebble challenge test server at the `management_url` will be configured to respond to
+/// HTTP-01 challenge requests for a provided token by returning the expected key auth.
+struct HttpChallengeProvisioner {
+    management_url: String,
+    client: HyperClient<hyper_rustls::HttpsConnector<HttpConnector>, Full<Bytes>>,
```

Comment:
All `ChallengeProvisioner` impls will need a `management_url` and a HTTP `client`, right? Feels like this state should not live on this type... Can we reuse the `PebbleEnvironment` to pass around all of the state here? Or wrap it in a `ChallengeContext` or something?

Reply:
I also added a commit at the end to build the provisioner from the challenge type. I tacked that on separately in case you don't like the approach.